i18n-helpers/src/bin/mdbook-xgettext.rs

```diff
@@ -109,17 +266,31 @@ fn main() -> anyhow::Result<()> {
         .config
         .get_renderer("xgettext")
         .ok_or_else(|| anyhow!("Could not read output.xgettext configuration"))?;
-    let path = cfg
-        .get("pot-file")
-        .ok_or_else(|| anyhow!("Missing output.xgettext.pot-file config value"))?
-        .as_str()
-        .ok_or_else(|| anyhow!("Expected a string for output.xgettext.pot-file"))?;
+    // The depth on which to split the output file. 0 (default) is to include
+    // all messages into a single POT file. Greater values will split POT files,
+    // digging into the sub-chapters within each chapter.
+    let depth = match cfg.get("depth") {
+        None => 0,
+        Some(depth) => depth
+            .as_integer()
+            .with_context(|| "Expected a number for output.xgettext.depth")?,
+    };
     fs::create_dir_all(&ctx.destination)
         .with_context(|| format!("Could not create {}", ctx.destination.display()))?;
-    let output_path = ctx.destination.join(path);
-    let catalog = create_catalog(&ctx).context("Extracting messages")?;
-    polib::po_file::write(&catalog, &output_path)
-        .with_context(|| format!("Writing messages to {}", output_path.display()))?;
+    let catalogs = create_catalogs(&ctx, depth).context("Extracting messages")?;
+
+    // Create a template file for each entry with the content from the respective catalog.
+    for (mut file_path, catalog) in catalogs {
+        // Template files should have the "pot" extension.
+        assert!(file_path.set_extension("pot"));
```

Comment:
Two small things:

- Asserts are normally only used in tests so I would not assert anything here.
- I would suggest letting the code that forms `file_path` in `create_catalogs` do this. That way the code would take full responsibility for forming a useful path.

Reply:
Fair point about not using asserts in the executable, I had it there since it feels a bit better than dropping the boolean. But no problem to remove it

And I moved the `set_extension` to the `create_catalogs` code to have a better separation of the caller and the function body